test/reset.spec.js

```diff
@@ -0,0 +1,369 @@
+import {
+  sendResetLink,
+  updatePassword,
+  resetPassword,
+  updatePasswordWithLink,
+  updatePasswordWithJwt,
+} from "../src/reset.js";
+
+const tenantId = "abcd9876";
+const customBaseUrl = "https://custom.example.com/api/v1/";
+
+jest.mock("axios");
+
+// Using `window.location.assign` rather than `window.location.href =` because
+// JSDOM throws an error "Error: Not implemented: navigation (except hash changes)"
+// JSDOM complains about this is because JSDOM does not implement methods like window.alert, window.location.assign, etc.
+// https://stackoverflow.com/a/54477957
+delete window.location;
+window.location = {
+  assign: jest.fn(),
+  origin: "https://example.com",
+  href: "https://example.com/login",
+};
+
+// Mock API response
+const mockResponse = {
+  data: {
+    tokens: {
+      access: { value: createAccessToken() },
+      id: { value: createIdToken() },
+      refresh: { value: createRefreshToken() },
+    },
+    nonce: "nonce-value",
+    redirectTo: "/dashboard",
+  },
+};
+
+describe("sendResetLink", () => {
+  beforeEach(() => {
+    Userfront.init(tenantId);
+  });
+
+  it(`error should respond with whatever the server sends`, async () => {
+    // Mock the API response
+    const mockResponse = {
+      response: {
+        data: {
+          error: "Bad Request",
+          message: `That's a silly link request.`,
+          statusCode: 400,
+        },
+      },
+    };
+    axios.post.mockImplementationOnce(() => Promise.reject(mockResponse));
+    expect(sendResetLink("email@example.com")).rejects.toEqual(
+      new Error(mockResponse.response.data.message)
+    );
+  });
+
+  it(`should respond with whatever the server sends when using custom baseUrl`, async () => {
+    Userfront.init(tenantId, {
+      baseUrl: customBaseUrl,
+    });
+
+    // Mock the API response
+    const mockResponse = {
+      response: {
+        data: {
+          error: "Bad Request",
+          message: `That's a silly link request.`,
+          statusCode: 400,
+        },
+      },
+    };
+    axios.post.mockImplementationOnce(() => Promise.reject(mockResponse));
+
+    const email = "email@example.com";
+    expect(sendResetLink(email)).rejects.toEqual(
+      new Error(mockResponse.response.data.message)
+    );
+
+    // Should have sent the proper API request
+    expect(axios.post).toHaveBeenCalledWith(`${customBaseUrl}auth/reset/link`, {
+      email,
+      tenantId,
+    });
+  });
+});
+
+describe("resetPassword", () => {
+  it("should be an alias for updatePassword()", () => {
+    expect(resetPassword).toEqual(updatePassword);
+  });
+});
+
+describe("updatePassword", () => {
+  beforeEach(() => {
+    Userfront.init(tenantId);
+  });
+
+  afterEach(() => {
+    window.location.assign.mockClear();
+  });
+
+  describe("No method set (method is inferred)", () => {
+    it("should call updatePasswordWithLink() if JWT access token is not present", async () => {
```

Comment:
As implemented the logic is different - it calls `updatePasswordWithLink()` if link credentials are present in the call or the current query string, whether or not a JWT access token is present in the store.

I think the test itself is good - the description should be more like `it("should call updatePasswordWithLink() if token & uuid are passed to updatePassword()")`.

Reply:
Good catch! This was my initial approach but I changed it to preserve backwards-compatibility. I'll update the descriptions.

Reply:
Thank you!